src/managers/k8s.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Kubernetes manager."""
+
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import ServiceAccount
+
+from common.utils import WithLogging
+
+
+class KubernetesManager(WithLogging):
```

Comment:
**praise** I like to abstract away all K8s operations into a proper manager. This could indeed be used in multiple charms.